internal/profile/profile.go

```diff
@@ -12,13 +14,33 @@ var (
 	ErrInvalidGrepIdx = errors.New("invalid grep index")
 )
 
+type SeverityGetter interface {
+	GetSeverity() string
+}
+
+func (r *Request) GetSeverity() string {
+	return r.IssueSeverity
+}
+
+func (r *Response) GetSeverity() string {
+	return r.IssueSeverity
+}
+
+func (a *Active) GetSeverity() string {
+	if len(a.GetSteps()) > 0 {
+		return a.GetSteps()[0].IssueSeverity
+	}
+	return ""
+}
```

Comment:
I don't think this approach is a good idea, cause you want to consider all the steps for filtering.

For instance, the first step may have a `Info` severity, even not have an issue/severity, but a later step may have a `Critical` severity. So, in such case, if you're filtering to see only `Critical`, you'll want to keep this profile, and not filter it.